boards/v2/qemu/qemu_leon3/board.yml

```diff
@@ -0,0 +1,5 @@
+board:
+  name: qemu_leon3
```

Comment:
I think the `qemu` should become a single board where one select the SoC as `qemu/leon3`.

Ref: https://discord.com/channels/720317445772017664/1192088982616539210/1192826204122787890 and following comments.

Reply:
That's going to be one huge mess of a directory for files and documentation

Reply:
See continued discussion and agreement on Discord.

Just want to point out that board with full identifier now becomes: `qemu_leon3/leon3`.